domains/transfer-agreements/TransferAgreements.API/API/Filters/AuditDotNetFilter.cs

```diff
@@ -0,0 +1,30 @@
+using System.Security.Claims;
+using System.Threading.Tasks;
+using API.Extensions;
+using Audit.Core;
+using Microsoft.AspNetCore.Mvc.Filters;
+
+namespace API.Filters;
+
+public class AuditDotNetFilter : IAsyncActionFilter
+{
+    public async Task OnActionExecutionAsync(ActionExecutingContext context, ActionExecutionDelegate next)
+    {
+        var httpContext = context.HttpContext;
+        var claimsPrincipal = httpContext.User;
+
+        var actorId = claimsPrincipal.FindFirstValue("actor") ?? claimsPrincipal.FindFirstValue("atr");
+        var actorName = claimsPrincipal.FindFirstValue("name");
```

Comment:
There are extension methods defined in HttpContextUserExtensions. Please use them

Reply:
I have changed it, so it makes use of the extension methods, now.